### ISSUE TITLE
Sync progress notifications: incorrect transferrableBytes amount

### ISSUE THREAD

**Issue by raudabaugh** [realm/realm-mobile-platform#149]
## Goals

Show users the progress of downloading a ~15MB synced Realm for the first time and trigger an event when complete.

## Expected Results

Client SDK gets accurate value (~15MB) of transferrableBytes from server.

## Actual Results

Client SDK transferrableBytes value seems to be stuck at ~3.4MB.

## Code Sample
```
self.downloadToken = SyncUser.current?.allSessions().first?.addProgressNotification(for: .download, mode: .reportIndefinitely) { progress in
    print("=================================PROGRESS=============================================")
    print("fraction transferred \(progress.fractionTransferred)")
    print("transferrable bytes \(progress.transferrableBytes)")
    print("transferred bytes \(progress.transferredBytes)")
}
```
## Client Log
```
2017-03-15 13:02:40.956 Thread[1712:20362] Sync: Opening Realm file: /Users/sam/Library/Developer/CoreSimulator/Devices/EFEC05F6-CF08-48D7-ABCF-1CFCED7921D4/data/Containers/Data/Application/17A5A0F3-4179-46C9-A40D-72B603F377DE/Documents/realm-object-server/5f5bae43dc284888edd4d73f46024d57/realms%3A%2F%2F${mydomain.com}%2F${realmpath}
2017-03-15 13:02:41.010 Thread[1712:20362] Sync: Connection[3]: Session[1]: Starting session for '/Users/sam/Library/Developer/CoreSimulator/Devices/EFEC05F6-CF08-48D7-ABCF-1CFCED7921D4/data/Containers/Data/Application/17A5A0F3-4179-46C9-A40D-72B603F377DE/Documents/realm-object-server/5f5bae43dc284888edd4d73f46024d57/realms%3A%2F%2F${mydomain.com}%2F${realmpath}'
2017-03-15 13:02:41.010 Thread[1712:20362] Sync: Connection[3]: Resolving '${mydomain.com}:443'
2017-03-15 13:02:41.012 Thread[1712:20362] Sync: Connection[3]: Connecting to endpoint '${serverip}:443' (1/1)
2017-03-15 13:02:41.023 Thread[1712:20362] Sync: Connection[3]: Connected to endpoint '${serverip}:443' (from '${clientip}:49696')
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 0
transferred bytes 0
2017-03-15 13:02:41.125 Thread[1712:20362] Sync: Connection[3]: Session[1]: Sending: BIND(server_path='/${realmpath}', signed_user_token_size=629, need_file_ident_pair=1)
2017-03-15 13:02:41.152 Thread[1712:20362] Sync: Connection[3]: Session[1]: Received: ALLOC(server_file_ident=7398100213763755185, client_file_ident=124, client_file_ident_secret=2244420286831987036)
2017-03-15 13:02:41.153 Thread[1712:20362] Sync: Connection[3]: Session[1]: Sending: IDENT(server_file_ident=7398100213763755185, client_file_ident=124, client_file_ident_secret=2244420286831987036, scan_server_version=0, scan_client_version=0, latest_server_version=0, latest_server_session_ident=0)
=================================PROGRESS=============================================
fraction transferred 0.0207394511352558
transferrable bytes 3391218
transferred bytes 70332
=================================PROGRESS=============================================
fraction transferred 0.0404020030561291
transferrable bytes 3391218
transferred bytes 137012
=================================PROGRESS=============================================
fraction transferred 0.0612974453426468
transferrable bytes 3391218
transferred bytes 207873
=================================PROGRESS=============================================
fraction transferred 0.0837256702459116
transferrable bytes 3391218
transferred bytes 283932
=================================PROGRESS=============================================
fraction transferred 0.103060316381902
transferrable bytes 3391218
transferred bytes 349500
=================================PROGRESS=============================================
fraction transferred 0.122415604069098
transferrable bytes 3391218
transferred bytes 415138
=================================PROGRESS=============================================
fraction transferred 0.141788584514472
transferrable bytes 3391218
transferred bytes 480836
=================================PROGRESS=============================================
fraction transferred 0.161205501975986
transferrable bytes 3391218
transferred bytes 546683
=================================PROGRESS=============================================
fraction transferred 0.180629201661468
transferrable bytes 3391218
transferred bytes 612553
=================================PROGRESS=============================================
fraction transferred 0.200001297468933
transferrable bytes 3391218
transferred bytes 678248
=================================PROGRESS=============================================
fraction transferred 0.219449177257257
transferrable bytes 3391218
transferred bytes 744200
=================================PROGRESS=============================================
fraction transferred 0.238786182427671
transferrable bytes 3391218
transferred bytes 809776
=================================PROGRESS=============================================
fraction transferred 0.258143239390685
transferrable bytes 3391218
transferred bytes 875420
=================================PROGRESS=============================================
fraction transferred 0.277796355173864
transferrable bytes 3391218
transferred bytes 942068
=================================PROGRESS=============================================
fraction transferred 0.297159014843634
transferrable bytes 3391218
transferred bytes 1007731
=================================PROGRESS=============================================
fraction transferred 0.316494840496836
transferrable bytes 3391218
transferred bytes 1073303
=================================PROGRESS=============================================
fraction transferred 0.33631928115503
transferrable bytes 3391218
transferred bytes 1140532
=================================PROGRESS=============================================
fraction transferred 0.355656286325444
transferrable bytes 3391218
transferred bytes 1206108
=================================PROGRESS=============================================
fraction transferred 0.374986804151193
transferrable bytes 3391218
transferred bytes 1271662
=================================PROGRESS=============================================
fraction transferred 0.394325578597424
transferrable bytes 3391218
transferred bytes 1337244
=================================PROGRESS=============================================
fraction transferred 0.41373571383497
transferrable bytes 3391218
transferred bytes 1403068
=================================PROGRESS=============================================
fraction transferred 0.433076257557019
transferrable bytes 3391218
transferred bytes 1468656
=================================PROGRESS=============================================
fraction transferred 0.452427122054672
transferrable bytes 3391218
transferred bytes 1534279
=================================PROGRESS=============================================
fraction transferred 0.47177651215581
transferrable bytes 3391218
transferred bytes 1599897
=================================PROGRESS=============================================
fraction transferred 0.491136517911853
transferrable bytes 3391218
transferred bytes 1665551
=================================PROGRESS=============================================
fraction transferred 0.510482959219962
transferrable bytes 3391218
transferred bytes 1731159
=================================PROGRESS=============================================
fraction transferred 0.529810823131984
transferrable bytes 3391218
transferred bytes 1796704
=================================PROGRESS=============================================
fraction transferred 0.550333243100267
transferrable bytes 3391218
transferred bytes 1866300
=================================PROGRESS=============================================
fraction transferred 0.569698556683764
transferrable bytes 3391218
transferred bytes 1931972
=================================PROGRESS=============================================
fraction transferred 0.58903585673348
transferrable bytes 3391218
transferred bytes 1997549
=================================PROGRESS=============================================
fraction transferred 0.608499660004164
transferrable bytes 3391218
transferred bytes 2063555
=================================PROGRESS=============================================
fraction transferred 0.627849050105301
transferrable bytes 3391218
transferred bytes 2129173
=================================PROGRESS=============================================
fraction transferred 0.647175144741506
transferrable bytes 3391218
transferred bytes 2194712
=================================PROGRESS=============================================
fraction transferred 0.666541637842215
transferrable bytes 3391218
transferred bytes 2260388
=================================PROGRESS=============================================
fraction transferred 0.685884540598688
transferrable bytes 3391218
transferred bytes 2325984
=================================PROGRESS=============================================
fraction transferred 0.705223609924222
transferrable bytes 3391218
transferred bytes 2391567
=================================PROGRESS=============================================
fraction transferred 0.724572115387451
transferrable bytes 3391218
transferred bytes 2457182
=================================PROGRESS=============================================
fraction transferred 0.743909415437167
transferrable bytes 3391218
transferred bytes 2522759
=================================PROGRESS=============================================
fraction transferred 0.765644379099191
transferrable bytes 3391218
transferred bytes 2596467
=================================PROGRESS=============================================
fraction transferred 0.785010577320597
transferrable bytes 3391218
transferred bytes 2662142
=================================PROGRESS=============================================
fraction transferred 0.804375596024791
transferrable bytes 3391218
transferred bytes 2727813
=================================PROGRESS=============================================
fraction transferred 0.823724986125929
transferrable bytes 3391218
transferred bytes 2793431
=================================PROGRESS=============================================
fraction transferred 0.843488386768412
transferrable bytes 3391218
transferred bytes 2860453
=================================PROGRESS=============================================
fraction transferred 0.864158836146777
transferrable bytes 3391218
transferred bytes 2930551
=================================PROGRESS=============================================
fraction transferred 0.883512059678853
transferrable bytes 3391218
transferred bytes 2996182
=================================PROGRESS=============================================
fraction transferred 0.902857321469749
transferrable bytes 3391218
transferred bytes 3061786
=================================PROGRESS=============================================
fraction transferred 0.923521578382752
transferrable bytes 3391218
transferred bytes 3131863
=================================PROGRESS=============================================
fraction transferred 0.945692668533842
transferrable bytes 3391218
transferred bytes 3207050
=================================PROGRESS=============================================
fraction transferred 0.968938888623498
transferrable bytes 3391218
transferred bytes 3285883
=================================PROGRESS=============================================
fraction transferred 0.988466680702922
transferrable bytes 3391218
transferred bytes 3352106
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3417658
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3483315
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3551362
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3619384
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3703480
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3770488
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3837425
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3906417
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3972026
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4037592
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4103137
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4168700
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4234313
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4299872
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4365513
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4432504
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4498183
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4574684
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4640286
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4710645
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4776253
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4841910
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4908483
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 4974743
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5041681
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5107324
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5172988
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5238569
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5304106
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5369703
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5441858
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5518265
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5586947
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5653026
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5722564
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5788359
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5853946
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 5925340
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6002405
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6067962
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6145607
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6213550
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6279192
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6345682
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6421175
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6488108
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6554923
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6620498
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6686859
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6752456
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6819624
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6888171
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 6953739
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7019366
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7085846
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7154835
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7220482
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7287597
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7354418
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7420016
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7485702
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7551267
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7617343
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7684102
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7752790
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7818346
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7883985
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 7950482
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8016206
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8082374
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8149494
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8217649
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8283947
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8357971
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8426959
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8508049
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8576803
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8663577
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8731204
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8799835
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8865381
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8931005
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 8996713
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9063341
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9128996
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9194545
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9260347
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9325979
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9391935
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9458976
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9525389
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9591664
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9660440
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9728141
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9797890
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9863450
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9929014
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 9994641
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10060794
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10126441
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10191998
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10257563
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10323355
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10388940
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10454523
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10521136
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10589138
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10655808
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10721642
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10798751
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10864394
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 10935433
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11001212
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11073861
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11139512
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11205196
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11270859
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11338144
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11411589
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11477264
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11545218
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11612737
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11679693
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11747791
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11813331
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11881754
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 11947323
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12015139
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12083123
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12148676
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12214217
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12281122
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12350898
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12419133
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12485117
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12550757
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12619413
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12685023
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12753368
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12818965
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12885313
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 12952448
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13032613
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13101487
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13169058
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13237759
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13304828
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13384718
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13458934
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13534392
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13604550
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13672398
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13742303
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13813255
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13878810
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 13954229
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14020434
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14087167
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14153658
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14220676
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14289094
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14356511
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14425272
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14493980
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14561922
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 14628893
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3391218
2017-03-15 13:06:58.583 Thread[1712:20362] Sync: Connection[3]: Reading failed: End of input
2017-03-15 13:06:58.584 Thread[1712:20362] Sync: Connection[3]: Connection closed due to error
2017-03-15 13:06:58.585 Thread[1712:20362] Sync: Connection[3]: Resolving '${mydomain.com}:443'
2017-03-15 13:06:58.615 Thread[1712:20362] Sync: Connection[3]: Connecting to endpoint '${serverip}:443' (1/1)
2017-03-15 13:06:58.625 Thread[1712:20362] Sync: Connection[3]: Connected to endpoint '${serverip}:443' (from '${clientip}:49734')
2017-03-15 13:06:58.748 Thread[1712:20362] Sync: Connection[3]: Session[1]: Sending: BIND(server_path='/${realmpath}', signed_user_token_size=629, need_file_ident_pair=0)
2017-03-15 13:06:58.748 Thread[1712:20362] Sync: Connection[3]: Session[1]: Sending: IDENT(server_file_ident=7398100213763755185, client_file_ident=124, client_file_ident_secret=2244420286831987036, scan_server_version=43661, scan_client_version=2, latest_server_version=43661, latest_server_session_ident=1858442172696120058)
=================================PROGRESS=============================================
fraction transferred 1.0
transferrable bytes 3391218
transferred bytes 3391218
```
The realm itself contains sensitive data so I would prefer to share privately.

## Version of Realm and Tooling

Realm Object Server Version: 1.2.1
Flavor: Developer

Server OS & Version: Ubuntu 16.04.2

Client SDK Version: RealmSwift 2.4.4

Client OS & Version: iOS 10.2

**Comment by raudabaugh** [comment]
possibly related to #116

**Comment by Montrazul** [comment]
Yes, it is related to #116 . The progress notification is not working for the first start. So you cant use it for an initial realm download. Im already waiting for a fix for that (or another solution) because it is really neccessary for us to have all the data downloaded before presenting the views.

**Comment by Montrazul** [comment]
Wait! An Update from my site! I just noticed that a new version is available (2.4.4). I updated my app to 2.4.4 and finally the initial sync is working. Im using the following code:

```
import Foundation
import UIKit
import RealmSwift

class SyncViewController: UIViewController {
    @IBOutlet weak var progressLabel: UILabel!;
    @IBOutlet weak var progressBar: UIProgressView!;
    var notificationTokenProgress: NotificationToken!;
    var user: SyncUser!;
    
    static func storyboardInstance() -> SyncViewController? {
        let storyboard = UIStoryboard(name: String(describing: NSStringFromClass(SyncViewController.classForCoder()).components(separatedBy: ".").last!), bundle: nil);
        
        return storyboard.instantiateInitialViewController() as? SyncViewController;
    }

    override func viewDidLoad() {
        super.viewDidLoad();
        
        self.triggerRealm();
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        self.removeNotificationBlock();
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    func triggerRealm() {
        logger.debug("trigger realm");
        
        let email = SecureManager.getSharedInstance().getLogged()!;
        
        self.addNotificationProgress(email: email);
    }
 
    func addNotificationProgress(email: String) {
        if (self.user != nil) {
            let session = self.user.session(for: URL(string: HelperUrls.URL_REALM)!)!;
            
            logger.info("start notification for: \(email)");
            
            self.notificationTokenProgress = session.addProgressNotification(for: .download, mode: .forCurrentlyOutstandingWork) { progress in
                DispatchQueue.main.async {
                    logger.debug("sync progress");
                    logger.debug("fraction transferred \(progress.fractionTransferred)");
                    logger.debug("transferrable bytes \(progress.transferrableBytes)");
                    logger.debug("transferred bytes \(progress.transferredBytes)");
                    
                    if (!progress.fractionTransferred.isNaN) {
                        self.progressBar.setProgress(Float(progress.fractionTransferred), animated: true);
                        self.progressLabel.text = "\(Int((progress.fractionTransferred * 100))) %";
                    }
                    
                    if (progress.isTransferComplete) {
                        if (self.user != nil) {
                            logger.debug("login progress completed: \(self.user.identity!)");
                            
                            self.user = nil;
                            
                            self.presentSplitView();
                        }
                    }
                }
            }
        }
    }
    
    func removeNotificationBlock() {
        self.notificationToken?.stop();
        self.notificationTokenProgress?.stop();
    }
    
    func presentSplitView() {
        if let vc = SplitViewController.storyboardInstance() {
            self.present(vc, animated: true, completion: nil);
        }
    }
}
```

notice that im using `.forCurrentlyOutstandingWork` and not `.reportIndefinitely`.

**Comment by austinzheng** [comment]
Yes, 2.4.4 shipped with a fix. If you see any additional problems, please open a new ticket, and as always, thanks for using Realm!
